deepspeed/runtime/zero/stage2.py

```diff
@@ -230,6 +230,11 @@ def __init__(self,
         # number of elements per partition in each group
         self.partition_size = []
 
+        #align nccl all-gather send buffers to 4-bye boundary
+        self.nccl_start_alignment_factor = 2  # 4-byte alignment/sizeof(fp16) = 2
+
+        assert (allgather_bucket_size % self.nccl_start_alignment_factor == 0), "allgather_bucket_size must be a multiple of nccl_start_alignment_factor"
```

Comment:
It would be useful for this assert message to include the `nccl_start_alignment_factor` value so that users can adjust the `allgather_bucket_size` appropriately